data_integration/notification/slack.py

```diff
@@ -1,91 +1,42 @@
-"""Slack notifications for failed node runs"""
+"""Slack notifications"""
 
 from data_integration import config
-from data_integration.logging import pipeline_events
-from data_integration import events
+from data_integration.notification.chat_room import ChatRoom
+import requests
+import os
 
 
-class Slack(events.EventHandler):
-    node_output: {tuple: {bool: [events.Event]}} = None
+class Slack(ChatRoom):
 
-    def handle_event(self, event: events.Event):
-        """
-        Send the output of a node to Slack when the node failed.
-        Args:
-            event: The current event of interest
-        """
-        import requests
+    def __init__(self):
+        super().__init__(chat_type="Slack", code_markup_start="```", code_markup_end="```",
+                         line_start='\n _', line_end=' _ ')
 
-        if isinstance(event, pipeline_events.Output):
-            key = tuple(event.node_path)
+    def create_error_text(self, node_path: []):
+        path = '/'.join(node_path)
+        text = '\n:baby_chick: Ooops, a hiccup in ' + '_ <' + config.base_url() + '/' + path \
+               + ' | ' + path + ' > _'
+        return text
 
-            if not self.node_output:
-                self.node_output = {}
+    def create_error_msg(self, text, log, error_log):
+        message = {'text': text}
+        attachments = [{'text': log, 'mrkdwn_in': ['text']},
```

Comment:
attachments = []
if log:
    attachments.append(...)
if error_log:
    attachments.append(...)

Reply:
Yes, you are right.